### ISSUE TITLE
[Editor Performance] Bad performance in large graphs

### ISSUE THREAD

**Issue by IliaTonev** [Siccity/xNode#125]
Hello hello! :)

I've been using xNode for couple of months in a project, and first of all thank you for making it free and updating it regularly!

I'm not really experienced with coding (I work with the graphs as an end user with code support that provides me with custom logic and nodes) however, I have stumbled upon an issue while working with it. Large graphs tend to be very laggy and hard to navigate. Hopefully, the gif and the profiler image can better describe the issue. 

![2019-04-01_14-37-19](https://user-images.githubusercontent.com/49155394/55328772-74e81700-548d-11e9-92db-f46a6e653ffd.gif)

![2019-04-01_15-00-32](https://user-images.githubusercontent.com/49155394/55329485-f8563800-548e-11e9-8149-e21daa562ac0.png)

P.S. 1 I'm running it on a pretty solid machine, so I don't suspect hardware to be the issue (I can provide specs if you need them)
P.S. 2 I've checked similar issue (reported by other users), however they seemed to be describing a different issue.

**Comment by Siccity** [comment]
Yeah this has been an issue since the inception of xNode. A big part of it is due to the very slow way Unity draws serialized properties.

**Comment by Siccity** [comment]
Can you tell me, is your graph laggy as well when you're zoomed up real close? I'm considering a solution which drastically cuts down the amount of detail on nodes when zoomed out

**Comment by IliaTonev** [comment]
Hey!

As you mentioned, the most drastic hits in performance come when I'm getting the overview of a large graph (I'm zoomed out). 

LoDs for graphs, that sounds cool! 😊

**Comment by GZLHZ** [comment]
@Siccity Hi,I have the same problem,I create 190nodes and drags and feel slow.Can the JobSystem help this project improve the performance? anyway,thank you for this project.

**Comment by miduh** [comment]
Same problem in 2022
